include/swift/AST/Types.h

```diff
@@ -2887,27 +2893,57 @@ class AnyFunctionType : public TypeBase {
 
     unsigned Bits; // Naturally sized for speed.
 
-    ExtInfo(unsigned Bits) : Bits(Bits) {}
+  public:
+    class Uncommon {
+      friend ExtInfo;
+      friend class AnyFunctionType;
+      friend class FunctionType;
+      // We preserve a full clang::Type *, not a clang::FunctionType *, so
+      // we can keep sugar in case we need to present an error to the user.
+      const clang::Type *ClangFunctionType;
+
+      bool empty() const { return !ClangFunctionType; }
+      Uncommon(const clang::Type *type) : ClangFunctionType(type) {}
+    };
+
+  private:
+    Uncommon Other;
+
+    static void assertIsFunctionType(const clang::Type *);
+
+    ExtInfo(unsigned Bits, Uncommon Other) : Bits(Bits), Other(Other) {
+      // TODO: Once we start serializing the Clang type, we should also
+      // assert that the pointer is non-null.
+      auto Rep = Representation(Bits & RepresentationMask);
+      if ((Rep == Representation::CFunctionPointer) && Other.ClangFunctionType)
+        assertIsFunctionType(Other.ClangFunctionType);
+    }
+
+    static const clang::Type *getClangFunctionType(const Uncommon *uncommon) {
+      return uncommon->ClangFunctionType;
+    }
```

Comment:
It was being used earlier, because `AnyFunctionType` was a friend of `ExtInfo::Uncommon`, but not `FunctionType` was not a friend. After I made `FunctionType` a friend, I forgot to delete this method.

I wasn't really sure how "private" the `Uncommon` field oughta' be 😐.